agent/agent.go

```diff
@@ -94,6 +101,97 @@ type outputUnit struct {
 	outputs []*models.RunningOutput
 }
 
+func (a *Agent) RunSingleInput(input_config *models.InputConfig, ctx context.Context) {
+	// validating if an input plugin is already running, and therefore shouldn't be run again
+	_, ok := a.inputStatus[input_config.Name]
```

Comment:
Since multiple goroutines could be accessing this map at the same time, you'll need to put locks around it's accesses, everywhere it's accessed in the code.

For some context: There can be issues if two things that are happening 'concurrently' are changing the same values at the same time, so the way that you prevent this is by putting shared locks around things, which basically makes it so only one 'concurrent' action can do something at a time (instead of them both trying to do it at the same time).

Take a look @: https://tour.golang.org/concurrency/9

Another approach is to use the built in, concurrency safe Map ```sync.Map```.